static/js/navigation.js

```diff
@@ -0,0 +1,172 @@
+var dropdownWindow = document.querySelector('.dropdown-window');
+var dropdownWindowOverlay = document.querySelector('.dropdown-window-overlay');
+
+navDropdowns.forEach(function(dropdown) {
+  dropdown.addEventListener('click', function(event) {
+    event.preventDefault();
+
+    var clickedDropdown = this;
+
+    dropdownWindow.classList.remove('slide-animation');
+    dropdownWindowOverlay.classList.remove('fade-animation');
+
+    navDropdowns.forEach(function(dropdown) {
+      var dropdownContent = document.getElementById(dropdown.id + "-content");
+
+      if (dropdown === clickedDropdown) {
+
+        if (dropdown.classList.contains('is-open')) {
+          closeMenu(dropdown, dropdownContent);
+        } else {
+          dropdown.classList.add('is-open');
+          dropdownContent.classList.remove('u-hide');
+
+          if (window.history.pushState) {
+            window.history.pushState(null, null, '#' + dropdown.id);
+          }
+        }
+      } else {
+        dropdown.classList.remove('is-open');
+        dropdownContent.classList.add('u-hide');
+      }
+    });
+  });
+});
+
+// Close the menu if browser back button is clicked
+window.addEventListener('hashchange', function(event) {
+  navDropdowns.forEach(function (dropdown) {
+    const dropdownContent = document.getElementById(dropdown.id + "-content");
+
+    if (dropdown.classList.contains('is-open')) {
+      closeMenu(dropdown, dropdownContent);
+    }
+  });
+})
+
+dropdownWindowOverlay.addEventListener('click', function(event) {
+  navDropdowns.forEach(function(dropdown) {
+    var dropdownContent = document.getElementById(dropdown.id + "-content");
+
+    if (dropdown.classList.contains('is-open')) {
+      closeMenu(dropdown, dropdownContent);
+    }
+  });
+});
+
+function closeMenu(dropdown, dropdownContent) {
+  dropdown.classList.remove('is-open');
+  dropdownWindow.classList.add('slide-animation');
+  dropdownWindowOverlay.classList.add('fade-animation');
+  if (window.history.pushState) {
+    window.history.pushState(null, null, window.location.href.split('#')[0]);
+  }
+}
+
+if (window.location.hash) {
+  var tabId = window.location.hash.split('#')[1];
+  var tab = document.getElementById(tabId);
+  var tabContent = document.getElementById(tabId + '-content');
+
+  if (tab) {
+    setTimeout(function() {
+      document.getElementById(tabId).click();
+    }, 0);
+  }
+}
+
+function closeMainMenu() {
+  var navigationLinks = document.querySelectorAll('.p-navigation__dropdown-link');
+
+  navigationLinks.forEach(function(navLink) {
+    navLink.classList.remove('is-open');
+  });
+
+  if (!dropdownWindowOverlay.classList.contains('fade-animation')) {
+    dropdownWindowOverlay.classList.add('fade-animation');
+  }
+
+  if (!dropdownWindow.classList.contains('slide-animation')) {
+    dropdownWindow.classList.add('slide-animation');
+  }
+}
+
+var origin = window.location.href;
+
+addGANavEvents('#canonical-products', 'www.ubuntu.com-nav-0-products');
```

Comment:
if these events are still needed they need to be updated.

Reply:
Good catch. Updated the GA events.